types/node/child_process.d.ts

```diff
@@ -157,89 +157,51 @@ declare module "child_process" {
-    ): ChildProcessByStdio<null, Readable, null>;
-    function spawn(
-        command: string,
-        options: SpawnOptionsWithStdioTuple<StdioNull, StdioNull, StdioPipe>,
-    ): ChildProcessByStdio<null, null, Readable>;
-    function spawn(
-        command: string,
-        options: SpawnOptionsWithStdioTuple<StdioNull, StdioNull, StdioNull>,
-    ): ChildProcessByStdio<null, null, null>;
-
-    function spawn(command: string, options: SpawnOptions): ChildProcess;
-
-    // overloads of spawn with 'args'
-    function spawn(command: string, args?: ReadonlyArray<string>, options?: SpawnOptionsWithoutStdio): ChildProcessWithoutNullStreams;
+    interface SpawnFunc<Options, Return> {
+        (command: string, options: Options): Return;
+        (command: string, args: ReadonlyArray<string>, options: Options): Return;
+    }
 
-    function spawn(
-        command: string,
-        args: ReadonlyArray<string>,
-        options: SpawnOptionsWithStdioTuple<StdioPipe, StdioPipe, StdioPipe>,
-    ): ChildProcessByStdio<Writable, Readable, Readable>;
-    function spawn(
-        command: string,
-        args: ReadonlyArray<string>,
-        options: SpawnOptionsWithStdioTuple<StdioPipe, StdioPipe, StdioNull>,
-    ): ChildProcessByStdio<Writable, Readable, null>;
-    function spawn(
-        command: string,
-        args: ReadonlyArray<string>,
-        options: SpawnOptionsWithStdioTuple<StdioPipe, StdioNull, StdioPipe>,
-    ): ChildProcessByStdio<Writable, null, Readable>;
-    function spawn(
-        command: string,
-        args: ReadonlyArray<string>,
-        options: SpawnOptionsWithStdioTuple<StdioNull, StdioPipe, StdioPipe>,
-    ): ChildProcessByStdio<null, Readable, Readable>;
-    function spawn(
-        command: string,
-        args: ReadonlyArray<string>,
-        options: SpawnOptionsWithStdioTuple<StdioPipe, StdioNull, StdioNull>,
-    ): ChildProcessByStdio<Writable, null, null>;
-    function spawn(
-        command: string,
-        args: ReadonlyArray<string>,
-        options: SpawnOptionsWithStdioTuple<StdioNull, StdioPipe, StdioNull>,
-    ): ChildProcessByStdio<null, Readable, null>;
-    function spawn(
-        command: string,
-        args: ReadonlyArray<string>,
-        options: SpawnOptionsWithStdioTuple<StdioNull, StdioNull, StdioPipe>,
-    ): ChildProcessByStdio<null, null, Readable>;
-    function spawn(
-        command: string,
-        args: ReadonlyArray<string>,
-        options: SpawnOptionsWithStdioTuple<StdioNull, StdioNull, StdioNull>,
-    ): ChildProcessByStdio<null, null, null>;
+    interface SpawnFuncOptParams<Options, Return> {
+        (command: string, options?: Options): Return;
+        (command: string, args?: ReadonlyArray<string>, options?: Options): Return;
+    }
 
-    function spawn(command: string, args: ReadonlyArray<string>, options: SpawnOptions): ChildProcess;
+    const spawn:
+        SpawnFuncOptParams<SpawnOptionsWithoutStdio, ChildProcessWithoutNullStreams> &
```

Comment:
Can't say I'm a fan of this, this drastically increases the complexity and removes overload selection.